examples/caldav-get-lib.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python3
+from icalendar import Calendar
+import caldav
+import json
+from os import environ as env
+
+username = env["CALDAV_USERNAME"]
+password = env["CALDAV_PASSWORD"]
+url      = env["CALDAV_URL"]
+caldav_url = f"https://{url}/{username}/"
+headers = {}
+
+
+def fetch_and_print():
+    with caldav.DAVClient(
+        url=caldav_url,
+        username=username,
+        password=password,
+        headers=headers,  # Optional parameter to set HTTP headers on each request if needed
+    ) as client:
+        print_calendars_demo(client.principal().calendars())
+
+
+def print_calendars_demo(calendars):
+    if not calendars:
+        return
+    events = []
+    for calendar in calendars:
+        for eventraw in calendar.events():
+            for component in Calendar.from_ical(eventraw._data).walk():
```

Comment:
Use rather `eventraw.icalendar_instance.walk()`.

Also, `eventraw._data` is internal and not supposed to be used directly.  `eventraw.data` is more appropriate.

Tidbit: the icalendar instance created through `eventraw.icalendar_instance` is still linked with the `eventraw` object - so if you edit the icalendar instance and then later call `eventraw.data`, then it will include the changes done on the icalendar object.

Reply:
done